experimental/report/diagnostic.go

```diff
@@ -198,6 +238,8 @@ type snippet struct {
 	// Whether this is a "primary" snippet, which is used for deciding whether or not
 	// to mark the snippet with the same color as the overall diagnostic.
 	primary bool
+
+	edits []Edit
```

Comment:
nit: the struct's Go doc should likely be updated to describe the alternate rendering behavior when this is non-empty.

Also, does this really need to be a slice? I think only the thousands-separator diagnostic uses multiple edits, but it could just as easily suggest a single edit that replaces the entire numeric literal with a value that has no underscores.

Reply:
Yes, for errors where we specify missing brackets. It's not strictly necessary, but I don't really want to give up the flexibility.